### ISSUE TITLE
Call to undefined method in MW 1.36

### ISSUE THREAD

**Issue by Ostrzyciel** [SemanticMediaWiki/SemanticExtraSpecialProperties#159]
### Setup and configuration

- SMW version: master [ac844d69](https://github.com/SemanticMediaWiki/SemanticMediaWiki/commit/ac844d69e015bf54fb1cbb16a687c27c9e4056b8)
- SESP version: master [b82b4e10](https://github.com/SemanticMediaWiki/SemanticExtraSpecialProperties/commit/b82b4e10accdaccc3a9841bc1a9f92c51b029a22)
- MW version: 1.36.1
- PHP version: 7.4.22
- DB: MariaDB 10.6.3

### Issue

Running the `rebuildStore.php` script with SESP installed and the `_CUSER` property enabled, a stack trace is emitted:

```
Error: Call to undefined method MediaWiki\User\UserIdentityValue::getUserPage()
Backtrace:
from /var/www/html/extensions/SemanticExtraSpecialProperties/src/PropertyAnnotators/CreatorPropertyAnnotator.php(63)
#0 /var/www/html/extensions/SemanticExtraSpecialProperties/src/PropertyAnnotators/DispatchingPropertyAnnotator.php(73): SESP\PropertyAnnotators\CreatorPropertyAnnotator->addAnnotation(SMW\DIProperty, SMW\SemanticData)
#1 /var/www/html/extensions/SemanticExtraSpecialProperties/src/ExtraPropertyAnnotator.php(72): SESP\PropertyAnnotators\DispatchingPropertyAnnotator->addAnnotation(SMW\DIProperty, SMW\SemanticData)
#2 /var/www/html/extensions/SemanticExtraSpecialProperties/src/HookRegistry.php(176): SESP\ExtraPropertyAnnotator->addAnnotation(SMW\SemanticData)
#3 /var/www/html/includes/HookContainer/HookContainer.php(338): SESP\HookRegistry->SESP\{closure}(SMW\SPARQLStore\SPARQLStore, SMW\SemanticData)
#4 /var/www/html/includes/HookContainer/HookContainer.php(137): MediaWiki\HookContainer\HookContainer->callLegacyHook(string, array, array, array)
#5 /var/www/html/includes/Hooks.php(137): MediaWiki\HookContainer\HookContainer->run(string, array, array)
#6 /var/www/html/extensions/SemanticMediaWiki/src/Store.php(230): Hooks::run(string, array)
#7 /var/www/html/extensions/SemanticMediaWiki/src/DataUpdater.php(415): SMW\Store->updateData(SMW\SemanticData)
#8 /var/www/html/extensions/SemanticMediaWiki/src/DataUpdater.php(294): SMW\DataUpdater->updateData()
#9 /var/www/html/extensions/SemanticMediaWiki/src/DataUpdater.php(198): SMW\DataUpdater->runUpdate()
#10 /var/www/html/extensions/SemanticMediaWiki/src/ParserData.php(454): SMW\DataUpdater->doUpdate()
#11 /var/www/html/extensions/SemanticMediaWiki/src/MediaWiki/Jobs/UpdateJob.php(303): SMW\ParserData->updateStore()
#12 /var/www/html/extensions/SemanticMediaWiki/src/MediaWiki/Jobs/UpdateJob.php(216): SMW\MediaWiki\Jobs\UpdateJob->updateStore(SMW\ParserData)
#13 /var/www/html/extensions/SemanticMediaWiki/src/MediaWiki/Jobs/UpdateJob.php(137): SMW\MediaWiki\Jobs\UpdateJob->parse_content()
#14 /var/www/html/extensions/SemanticMediaWiki/src/MediaWiki/Jobs/UpdateJob.php(94): SMW\MediaWiki\Jobs\UpdateJob->doUpdate()
#15 /var/www/html/extensions/SemanticMediaWiki/src/Maintenance/DistinctEntityDataRebuilder.php(193): SMW\MediaWiki\Jobs\UpdateJob->run()
#16 /var/www/html/extensions/SemanticMediaWiki/src/Maintenance/DistinctEntityDataRebuilder.php(170): SMW\Maintenance\DistinctEntityDataRebuilder->doUpdate(SMW\MediaWiki\JobFactory, Title)
#17 /var/www/html/extensions/SemanticMediaWiki/src/Maintenance/DataRebuilder.php(267): SMW\Maintenance\DistinctEntityDataRebuilder->doRebuild()
#18 /var/www/html/extensions/SemanticMediaWiki/src/Maintenance/DataRebuilder.php(318): SMW\Maintenance\DataRebuilder->rebuildFromSelection(array)
#19 /var/www/html/extensions/SemanticMediaWiki/src/Maintenance/DataRebuilder.php(222): SMW\Maintenance\DataRebuilder->rebuildAll()
#20 /var/www/html/extensions/SemanticMediaWiki/maintenance/rebuildData.php(198): SMW\Maintenance\DataRebuilder->rebuild()
#21 /var/www/html/maintenance/doMaintenance.php(112): SMW\Maintenance\rebuildData->execute()
#22 /var/www/html/extensions/SemanticMediaWiki/maintenance/rebuildData.php(294): require_once(string)
#23 {main}
```

I do realize that this is an unsupported configuration and I'll happily contribute to fix this problem, but I am a bit confused by the current compatibility status of the extension. With which MW versions should the master branch be compatible with? Having functioning CI would help greatly as well. :)

**Comment by Seb35** [comment]
This is solved by f952ca13914630f9fb3275269b9b5b280762e0ef (provided in SESP 3.0.0+), I confirmed it works on MW 1.38 + SMW 4.0.2 + SESP 3.0.1, so **this issue can be closed**.

Also, I just checked other occurences of getUserPage in SESP (2 in `/src`, 3 in `/tests`), it seems they really have a User object (not a UserIdentity) so these other occurences should be fine for now.

**Comment by kghbln** [comment]
Thanks for your assessment!
